src/interrupt/mod.rs

```diff
@@ -0,0 +1,13 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+use std::result::Result;
+
+/// Abstraction for a simple, push-button like interrupt mechanism.
+pub trait Trigger {
+    /// Underlying type for the potential error conditions returned by `Self::trigger`.
+    type E;
```

Comment:
It would be better and more typical to use `Error` for the generic error type (this is used https://doc.rust-lang.org/std/convert/trait.TryFrom.html for instance).